### PR TITLE
Close #30: Switch last few routes to class-based views

### LINKED ISSUE #30: Switch to class based views
As seen here: http://flask.pocoo.org/docs/0.12/views/

### PR DESCRIPTION
**Pull request by manterolat**
Adds class-based views for:
 - Index
 - Team list
 - Rename team
 - Update team (new route, fixes team editing)

### CODE REVIEW

**Comment by andrewsosa** [comment]
@manterolat just to be sure, you tested these right? I'll take a look myself when I get home today.

**Comment by manterolat** [comment]
@andrewsosa Yeah, using a virtual machine. That's how I found that the Team Update form wasn't working.

File: app/views/team/rename.py [hunk 1]
```diff
@@ -5,36 +5,35 @@
 from app import app
 from app.util.auth import get_account
 from app.util.team import rename_team, set_division
+from .team import TeamView
 
+class RenameTeamView(TeamView):
```
Comment (andrewsosa) [inline app/views/team/rename.py]: @manterolat -- It looks like we never actually use the RenameTeamView functionality, and only just use the UpdateTeamView. Unless  you see a spot where we use RenameTeamView, let's just go ahead and remove it

File: app/views/team/update.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# views.team.update
```
Comment (andrewsosa) [inline app/views/team/update.py]: Good job with this one; I didn't realize this wasn't coming I wrote at first.

File: app/views/team/__init__.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
-from . import rename
+from .rename import RenameTeamView as RenameView
```
Comment (andrewsosa) [inline app/views/team/__init__.py]: See comment about removing RenameTeamView

File: app/views/__init__.py [hunk 4]
```diff
@@ -51,10 +60,12 @@ def __init__(self, url, view):
     Route('/account', account.EditAccountView.as_view('account')),
     Route('/account/profile', account.ProfileView.as_view('profile')),
     Route('/account/team', team.TeamView.as_view('team')),
+    Route('/account/team/update', team.UpdateView.as_view('team_update')),
     Route('/account/team/create', team.CreateView.as_view('team_create')),
     Route('/account/team/add', team.AddView.as_view('team_add_member')),
     Route('/account/team/leave', team.LeaveView.as_view('team_leave')),
     Route('/account/team/remove', team.RemoveView.as_view('team_remove')),
+    Route('/account/team/rename', team.RenameView.as_view('team_rename')),
```
Comment (andrewsosa) [inline app/views/__init__.py]: See comment about removing RenameTeamView

**Pull request merged**